packages/manager/apps/pci-private-network/src/api/data/network.ts

```diff
@@ -30,3 +43,140 @@ export const deleteNetwork = async (
 
   return data;
 };
+export const associateGatewayToNetworkCall = async (
+  projectId: string,
+  region: string,
+  gatewayId: string,
+  subnetId: string,
+) => {
+  const { data } = await v6.post(
+    `/cloud/project/${projectId}/region/${region}/gateway/${gatewayId}/interface`,
+    {
+      subnetId,
+    },
+  );
+
+  return data;
+};
+
+export const enableSnatOnGatewayCall = async (
+  projectId: string,
+  region: string,
+  gatewayId: string,
+) => {
+  const { data } = await v6.post(
+    `/cloud/project/${projectId}/region/${region}/gateway/${gatewayId}/expose`,
+  );
+
+  return data;
+};
+
+export const getCreatedSubnet = async (
+  projectId: string,
+  region: string,
+  networkId: string,
+) => {
+  const { data } = await v6.get(
+    `/cloud/project/${projectId}/region/${region}/network/${networkId}/subnet`,
+  );
+
+  return data;
+};
+
+export const checkPrivateNetworkCreationStatus = async (
+  projectId: string,
+  operationId: string,
+) => {
+  const { data } = await v6.get<TNetworkCreationResponse>(
+    `/cloud/project/${projectId}/operation/${operationId}`,
+  );
+
+  return data;
+};
+
+export const createNetworkCall = async (
+  projectId: string,
+  region: string,
+  privateNetworkName: string,
+  subnet,
+  gateway,
+  vlanId = null,
+) => {
+  const { data } = await v6.post<TNetworkCreationResponse>(
+    `/cloud/project/${projectId}/region/${region}/network`,
+    {
+      name: privateNetworkName,
+      subnet,
+      ...(gateway && { gateway }),
+      ...(typeof vlanId === 'number' && { vlanId }),
+    },
+  );
+
+  return data;
+};
+
+const waitForNetworkCreation = async (projectId: string, networkId: string) => {
+  const waitFor = (ms = 1000) =>
+    new Promise((resolve) => {
+      setTimeout(resolve, ms);
+    });
+
+  const response: TNetworkCreationResponse = await checkPrivateNetworkCreationStatus(
+    projectId,
+    networkId,
+  );
+
+  if (response.status !== 'completed') {
```

Comment:
i just checked the possible status in api are : completed┃created┃in-error┃in-progress┃unknown

we should handle the "in-error" status as well such as :

```ts
if (reponse.status === 'in-error') {
    return Promise.reject(response);
}
```

Reply:
will update the operation hook as well, may be it's too late for this branch but it could be used for future ones